src/binaryen-c.h

```diff
@@ -338,24 +385,85 @@ typedef void* BinaryenFunctionRef;
 //            0 (and written $0), and if you also have 2 vars they will be
 //            at indexes 1 and 2, etc., that is, they share an index space.
 BinaryenFunctionRef BinaryenAddFunction(BinaryenModuleRef module, const char* name, BinaryenFunctionTypeRef type, BinaryenType* varTypes, BinaryenIndex numVarTypes, BinaryenExpressionRef body);
+// Gets the function at the specified index. Returns `NULL` when there are no more functions.
+BinaryenFunctionRef BinaryenGetFunctionAt(BinaryenModuleRef module, BinaryenIndex index);
+// Gets the name of the specified function.
+const char* BinaryenFunctionGetName(BinaryenFunctionRef func);
+// Sets the name of the specified function.
+void BinaryenFunctionSetName(BinaryenFunctionRef func, const char* newName);
+// Gets the body of the specified function.
+BinaryenExpressionRef BinaryenFunctionGetBody(BinaryenFunctionRef func);
+// Sets the body of the specified function.
+void BinaryenFunctionSetBody(BinaryenFunctionRef func, BinaryenExpressionRef expr);
+// Gets the parameter type at the specified index of the specified function. Returns `BinaryenUndefined()`
+// when there are no more parameters.
+BinaryenType BinaryenFunctionGetParamAt(BinaryenFunctionRef func, BinaryenIndex index);
+// Gets the result type of the specified function.
+BinaryenType BinaryenFunctionGetResult(BinaryenFunctionRef func);
 
 // Imports
 
 typedef void* BinaryenImportRef;
 
 BinaryenImportRef BinaryenAddImport(BinaryenModuleRef module, const char* internalName, const char* externalModuleName, const char *externalBaseName, BinaryenFunctionTypeRef type);
-void BinaryenRemoveImport(BinaryenModuleRef module, const char* internalName);
+// Removes the import of the specified internal name from the module.
+void BinaryenRemoveImportByName(BinaryenModuleRef module, const char* internalName);
+// Gets the import at the specified index from the specified module. Returns `NULL` when there are no more imports.
+BinaryenImportRef BinaryenGetImportAt(BinaryenModuleRef module, BinaryenIndex index);
+// Gets the internal name of the specified import. This is the name referenced by other internal elements.
+const char* BinaryenImportGetInternalName(BinaryenImportRef import);
```

Comment:
can use just `Name`, as that's the field name in the code. But the comment might mention it is the internal name.